aten/src/ATen/core/op_registration/infer_schema.h

```diff
@@ -18,63 +18,69 @@ void checkStaticTypes() {
  // Give nice error messages for some of the common error cases.
  // Use a LOUD ERROR MESSAGE SO USERS SEE THE STATIC_ASSERT
  static_assert(
-     !std::is_integral<T>::value || std::is_same<T, int64_t>::value,
-     "INVALID TYPE: Only int64_t is supported as an integral argument type");
+     !std::is_integral<T>::value || std::is_same<T, int64_t>::value || std::is_same<T, bool>::value,
+     "INVALID TYPE: Only int64_t and bool are supported as an integral argument type");
  static_assert(
      !std::is_same<T, float>::value,
      "INVALID TYPE: float is not supported as an argument type, use double instead");
 }
 
-template <typename First, typename Second, typename... Rest>
-void checkStaticTypes() {
- checkStaticTypes<First>();
- checkStaticTypes<Second, Rest...>();
-}
-
 template <typename... Ts, size_t... Is>
 ::std::vector<Argument> createArgumentVectorFromTypes(guts::index_sequence<Is...>) {
-  checkStaticTypes<guts::decay_t<Ts>...>();
+  // Check types for common errors
```

Comment:
I don't know what changed here or why. I also am noticing a ton of useful comments have been dropped like this:

```/// Unpack a tuple return type into a vector of return types, one per tuple	template<class... ParameterTypes>
/// element.
```
 This code is extremely arcane template meta-programming stuff and should remain commented.

Reply:
I'll re-add the comments.

These are the main changes:
- `createReturns` now works correctly for kernels returning `void`. Before, it didn't support this case.
- The old `checkStaticTypes` implementation only worked if the argument list had at least one type (since `checkStaticTypes` required the `First` template parameter to be set). The new implementation with the initializer list also works for empty argument lists, which is for example needed for kernels with a `void` return.